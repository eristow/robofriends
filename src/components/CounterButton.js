import React, { useState } from 'react';

const CounterButton = ({ color }) => {
  const [count, setCount] = useState(0);

  const updateCount = () => setCount(count + 1);

  return (
    <button id="counter" color={color} onClick={updateCount}>
      Count: {count}
    </button>
  );
};

export default CounterButton;
