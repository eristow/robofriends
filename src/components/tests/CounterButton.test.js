import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from '../CounterButton';

describe('CounterButton', () => {
  const mockColor = 'blue';

  it('should render the CounterButton component', () => {
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
  });

  it('should increment count when clicked', () => {
    const button = shallow(<CounterButton color={mockColor} />);
    button.find('[id="counter"]').simulate('click');
    expect(button.text()).toEqual('Count: 1');
    expect(button.props().color).toEqual('blue');
  });
});
