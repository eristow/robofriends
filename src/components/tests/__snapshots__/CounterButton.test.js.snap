// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton should render the CounterButton component 1`] = `
<button
  color="blue"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
